### PR TITLE
[AMORO-2279] Retry failed tasks when restarting AMS

### LINKED ISSUE #2279: [Bug]: Optimizing status blocked at minor/major when restarting AMS
### What happened?

The scenario is that the failed task has entered the retry queue, but has not yet been consumed by the optimizer, at which point ams restarts the recovery, and the failed task will no longer be executed

### Affects Versions

master/0.6.0/0.5.x

### What engines are you seeing the problem on?

AMS

### How to reproduce

1. plan some optimization tasks that optimizer will not be execute sucessfully
2. start optimizer and execute the tasks
3. restart AMS

### Relevant log output

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Close #2279

## Brief change log

- Avoding optimizing status blocked

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by XBaith** [comment]
cc @wangtaohz

**Review by wangtaohz — APPROVED** [review]
LGTM.

Thanks for you contribution 👍

**Comment by XBaith** [comment]
> LGTM.

Can you cherry-pick this PR to branch 0.6.x?

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java [hunk 1]
```diff
@@ -122,15 +122,23 @@ private void initTableRuntime(TableRuntimeMeta tableRuntimeMeta) {
...
+                  switch (taskRuntime.getStatus()) {
+                    case SCHEDULED:
+                    case ACKED:
+                      executingTaskMap.put(taskId, taskRuntime);
+                      break;
+                    case PLANNED:
+                      taskQueue.offer(taskRuntime);
+                      break;
+                    case FAILED:
+                      retryQueue.offer(taskRuntime);
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java:138]: Using com.netease.arctic.server.optimizing.OptimizingQueue#retryTask

Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java:138]: sure

**Comment by wangtaohz** [comment]
> Can you cherry-pick this PR to branch 0.6.x?

Ok, I will.

**Comment by shidayang** [comment]
LGTM

**Pull request merged**